bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java

```diff
@@ -646,9 +656,18 @@ public ServerConfiguration setJournalDirsName(String[] journalDirs) {
      *
      * @return ledger dir names, if not provided return null
      */
+    public String[] getLedgerDirWithoutDefault() {
+        return this.getStringArray(LEDGER_DIRS);
+    }
+
+    /**
+     * Get dir names to store ledger data
+     *
+     * @return ledger dir names, if not provided return null
+     */
     public String[] getLedgerDirNames() {
         String[] ledgerDirs = this.getStringArray(LEDGER_DIRS);
-        if (null == ledgerDirs) {
+        if ((null == ledgerDirs) || (0 == ledgerDirs.length)) {
```

Comment:
@sijie 
just to double check:
I wonder if this change can be dangerous, as if the user configures and 'empty' string for the ledgersDirs it will fallback to /tmp without knowing.

I think that the 'cookie' on ZK will prevent the bookie from starting, so this change is safe
Am I correct ?